exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/statistics/StatisticsAggBatch.java

```diff
@@ -0,0 +1,256 @@
+    keyOutputIds.add(id);
+
+    if (collector.hasErrors()) {
+      throw new SchemaChangeException("Failure while materializing expression. "
+          + collector.toErrorString());
+    }
+  }
+
+  private void addMapVector(String name, MapVector parent, LogicalExpression expr,
+      List<LogicalExpression> valueExprs) throws SchemaChangeException {
+    ErrorCollector collector = new ErrorCollectorImpl();
+
+    LogicalExpression mle = ExpressionTreeMaterializer.materialize(expr, incoming, collector,
+        context.getFunctionRegistry());
+
+    Class<? extends ValueVector> vvc =
+        TypeHelper.getValueVectorClass(mle.getMajorType().getMinorType(), mle.getMajorType().getMode());
+    ValueVector vv = parent.addOrGet(name, mle.getMajorType(), vvc);
+
+    TypedFieldId pfid = container.getValueVectorId(SchemaPath.getSimplePath(parent.getField().getPath()));
+    assert pfid.getFieldIds().length == 1;
+    TypedFieldId.Builder builder = TypedFieldId.newBuilder();
+    builder.addId(pfid.getFieldIds()[0]);
+    TypedFieldId id = FieldIdUtil.getFieldIdIfMatches(parent, builder, true,
+        SchemaPath.getSimplePath(vv.getField().getPath()).getRootSegment());
+
+    valueExprs.add(new ValueVectorWriteExpression(id, mle, true));
+
+    if (collector.hasErrors()) {
+      throw new SchemaChangeException("Failure while materializing expression. "
+          + collector.toErrorString());
+    }
+  }
+
+  private StreamingAggregator codegenAggregator(List<LogicalExpression> keyExprs,
+      List<LogicalExpression> valueExprs, List<TypedFieldId> keyOutputIds)
+          throws SchemaChangeException, ClassTransformationException, IOException {
+    ClassGenerator<StreamingAggregator> cg =
+        CodeGenerator.getRoot(StreamingAggTemplate.TEMPLATE_DEFINITION, context.getFunctionRegistry(),
+            context.getOptions());
+
+    LogicalExpression[] keyExprsArray = new LogicalExpression[keyExprs.size()];
+    LogicalExpression[] valueExprsArray = new LogicalExpression[valueExprs.size()];
+    TypedFieldId[] keyOutputIdsArray = new TypedFieldId[keyOutputIds.size()];
+
+    keyExprs.toArray(keyExprsArray);
+    valueExprs.toArray(valueExprsArray);
+    keyOutputIds.toArray(keyOutputIdsArray);
+
+    setupIsSame(cg, keyExprsArray);
+    setupIsSameApart(cg, keyExprsArray);
+    addRecordValues(cg, valueExprsArray);
+    outputRecordKeys(cg, keyOutputIdsArray, keyExprsArray);
+    outputRecordKeysPrev(cg, keyOutputIdsArray, keyExprsArray);
+
+    cg.getBlock("resetValues")._return(JExpr.TRUE);
+    getIndex(cg);
+
+    container.buildSchema(SelectionVectorMode.NONE);
+    StreamingAggregator agg = context.getImplementationClass(cg);
+    agg.setup(oContext, incoming, this);
+    return agg;
+  }
+
+  private boolean isImplicitFileColumn(MaterializedField mf) {
+    return ImplicitColumnExplorer.initImplicitFileColumns(context.getOptions()).get(mf.getName()) != null;
+  }
+
+  protected StreamingAggregator createAggregatorInternal()
```

Comment:
The method is "create aggregator internal" which sounds like it creates an aggregator (generates a class.) At the end of this method, we do, in fact, generate code.

Should we make all this a bit clearer? Isn't all this stuff a per-schema setup? Or, are we trying to override an existing method? Probably not since no `@Override` on this method. Maybe rename "setupNewSchema" and break into parts?